lib/puppet/pops/loader/ruby_legacy_function_instantiator.rb

```diff
@@ -13,9 +14,15 @@ class Puppet::Pops::Loader::RubyLegacyFunctionInstantiator
   # @return [Puppet::Pops::Functions.Function] - an instantiated function with global scope closure associated with the given loader
   #
   def self.create(loader, typed_name, source_ref, ruby_code_string)
-    unless ruby_code_string.is_a?(String) && ruby_code_string =~ /Puppet\:\:Parser\:\:Functions.*newfunction/m
-      raise ArgumentError, _("The code loaded from %{source_ref} does not seem to be a Puppet 3x API function - no 'newfunction' call.") % { source_ref: source_ref }
+    # When func3x turned on, assert content by parsing, when turned off continue with (legacy) undefined behavior
+    if Puppet[:func3x_check]
```

Comment:
What about making this controlled via `:strict` instead (defaulting to not strict)?

Reply:
I considered that, It means you would have to give up all other kinds of strictness to turn this off. I suspect it is common to turn off strict to not see warnings.
If the protection we are adding were for something that is just flagging deprecated stuff then it would be ok, but now, if you have the illegal constructs they will just silently not work, so it is really not about "strictness" in the same sense.

I can be persuaded to use strict though if you feel that is better. 
If so, would error and warn do the validation and either fail or just log a warning, and ignore would not even do the validation?

Reply:
Sorry thumbsup meaning it's ok as is.